### ISSUE TITLE
Relax model observation rules

### ISSUE THREAD

**Issue by KeithWoods** [esp/esp-js#101]
Currently when you add a model to a router the router won't yield this model to model observers until the model has processed an initial event for that model

E.g no model yield without initial event :
```
class Model { /* assume some event subscriptions, either directly on the router or via decorators */ }
router.addModel("aModelId", new Model());
router.getModelObservable("aModelId").observe(model => { 
  /* won't yield until first event is published */ 
});
```
However once an initial event is published the model's observable stream always yields the model:
```
class Model { /* assume some event subscriptions, either directly on the router or via decorators */ }
router.addModel("aModelId", new Model());
router.publishEvent("aModelId", "andEvent", { /* payload*/} );
router.getModelObservable("aModelId").observe(model => { 
  /* will immediately yield */ 
});
```
This was by design. The intent was it gives you explicit control over when you start up you model. I.e. you control when you first publish an event and thus when your model's observable stream yields. 

As it turns out this is non-discoverable. If you add a model to the router and observe it via `getModelObservable` this stream should imminently yield if the model is registered. Additionally some models don't really require an initial event, their default state is sufficient for consumption after the models constructor is invoked.

This is a breaking changes as it would now be possible for an observer (i.e. a view) to receive a model that in a prior implementation may have relied on an initial event to complete model setup. As a side note, after a model constructor is called, it's state should be sufficient for observer consumption (i.e. good old OO design). If you do rely on an initial event you can do one of the following:
* initialising your model before you register it with the router.
* publish an event after you call `addModel`.

**In summary**
This change would ensure `getModelObservable` would always yield when the model is registered with the model, even if no event for the model has yet been published. 

Proposed change:
```
class Model { /* assume some event subscriptions, either directly on the router or via decorators */ }
router.addModel("aModelId", new Model());
router.getModelObservable("aModelId").observe(model => { 
  /* would yield  */ 
});
```

**Comment by KeithWoods** [comment]
closed via #109
